### PR TITLE
Add alertmanager route and receiver for mimir heartbeat

### PR DESCRIPTION
**Pull request by QuentinBisson**
## Description

Towards https://github.com/giantswarm/giantswarm/issues/30293, this PR adds the `route` and `receiver` for mimir heartbeat. The heartbeat alert is added https://github.com/giantswarm/prometheus-rules/pull/1094 and the alert has a label `type: "mimir-heartbeat"` (for legacy we use type: heartbeat). This label will eventually not be needed but it is what it is right now.

This PR should be released before https://github.com/giantswarm/prometheus-rules/pull/1094.

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: files/templates/alertmanager/alertmanager.yaml [hunk 1]
```diff
@@ -82,6 +93,7 @@ route:
     - team="atlas"
     - type!="heartbeat"
     - alertname!~"Inhibition.*"
+    - alertname!="Heartbeat"
```
Comment (QuentinBisson) [inline files/templates/alertmanager/alertmanager.yaml:96]: I could have used type!=mimir-heartbeat but I feel like this is better

File: service/controller/resource/alerting/alertmanagerconfig/resource_test.go [hunk 2]
```diff
@@ -110,7 +110,53 @@ func TestRenderingOfAlertmanagerConfig(t *testing.T) {
...
+		}
+
+		err = runner.Run()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestRenderingOfAlertmanagerConfigWithMimirEnabled(t *testing.T) {
```
Comment (QuentinBisson) [inline service/controller/resource/alerting/alertmanagerconfig/resource_test.go:138]: This add new tests to be able to render alertmanager config when mimir is enabled

**Comment by QuantumEnigmaa** [comment]
Ok I think this answers my question asked here https://github.com/giantswarm/prometheus-rules/pull/1094

**Review by QuantumEnigmaa — APPROVED** [review]
LGTM

**Pull request merged**